src/python/foglamp/core/scheduler.py

```diff
@@ -6,50 +6,349 @@
 
 """FogLAMP Scheduler"""
 
+import time
+import datetime
+from enum import IntEnum
 import asyncio
-from asyncio.subprocess import Process
+import logging
+import collections
+import uuid
+import sys # TODO: Needed for logging delete me
+
+import aiopg.sa
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql as pg_types
 
 __author__ = "Terris Linenbach"
 __copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
 __license__ = "Apache 2.0"
 __version__ = "${VERSION}"
 
-# For Process methods, see https://docs.python.org/3/library/asyncio-subprocess.html#asyncio.asyncio.subprocess.Process
-# or upgrade to a version of Python that uses type annotations
-_processes = []  # type: List[Process]
 
+class Scheduler(object):
```

Comment:
1. "Control+C to quit" is already output by aiohttp
2. I think KeyboardInterrupt is already handled by asyncio (SIGINT handler) 
3. SystemExit, which I haven't heard about until now, needs more thought if it is different from SIGTERM or SIGINT. That is best handled via an asyncio exception handler. But I'm not sure if there is a good way to handle this case. Wait for tasks to exit? Proper shutdown requires writing end times to the database. There might not be time to do that. This might not be a case where a graceful shutdown is possible.
4. With SchedulerBase you might be thinking of the different types of schedules. That would be justified in the API layer which I haven't thought about yet. Internally I'll keep it a union-tuple matching the table.